DiversifyingParentBlockQueryTests.testRandom fails comparing floating-point numbers

### CI Link

https://gradle-enterprise.elastic.co/s/jpdhq7xggo2iw/tests/task/:server:test/details/org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests/testRandom?top-execution=1

### Repro line

./gradlew ":server:test" --tests "org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests.testRandom" -Dtests.seed=5664ABE90D7EA1BC -Dtests.locale=kl-GL -Dtests.timezone=Europe/Jersey -Druntime.java=25

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
  2> REPRODUCE WITH: ./gradlew ":server:test" --tests "org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests.testRandom" -Dtests.seed=5664ABE90D7EA1BC -Dtests.locale=kl-GL -Dtests.timezone=Europe/Jersey -Druntime.java=25
  2> java.lang.AssertionError: 
    Expected: <0.4109492F>
         but: was <0.41094914F>
        at __randomizedtesting.SeedInfo.seed([5664ABE90D7EA1BC:24288EE6BC1E17CF]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2922)
        at org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests.lambda$testRandom$2(DiversifyingParentBlockQueryTests.java:133)
        at org.elasticsearch.index.mapper.MapperServiceTestCase.withLuceneIndex(MapperServiceTestCase.java:418)
        at org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests.testRandom(DiversifyingParentBlockQueryTests.java:78)
  2> NOTE: leaving temporary files on disk at: /Users/hinmanm/es/elasticsearch/server/build/testrun/test/temp/org.elasticsearch.search.vectors.DiversifyingParentBlockQueryTests_5664ABE90D7EA1BC-001
  2> NOTE: test params are: codec=Asserting(Lucene103): {}, docValues:{}, maxPointsInLeafNode=815, maxMBSortInHeap=6.724381532660603, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=kl-GL, timezone=Europe/Jersey
  2> NOTE: Mac OS X 26.2 aarch64/Oracle Corporation 25.0.1 (64-bit)/cpus=12,threads=1,free=469877504,total=536870912
  2> NOTE: All tests run in this JVM: [DiversifyingParentBlockQueryTests]
```

--- a/server/src/test/java/org/elasticsearch/search/vectors/DiversifyingParentBlockQueryTests.java
+++ b/server/src/test/java/org/elasticsearch/search/vectors/DiversifyingParentBlockQueryTests.java
@@ -22,12 +22,14 @@ import org.elasticsearch.xcontent.XContentType;
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 import java.util.TreeMap;
 
 import static org.apache.lucene.index.VectorSimilarityFunction.EUCLIDEAN;
+import static org.hamcrest.Matchers.closeTo;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
 
@@ -72,7 +74,7 @@ public class DiversifyingParentBlockQueryTests extends MapperServiceTestCase {
         List<TreeMap<Float, String>> expectedTopDocs = new ArrayList<>();
         for (int i = 0; i < numQueries; i++) {
             queries[i] = randomVector(dims);
-            expectedTopDocs.add(new TreeMap<>((o1, o2) -> -Float.compare(o1, o2)));
+            expectedTopDocs.add(new TreeMap<>(Comparator.reverseOrder()));
         }
 
         withLuceneIndex(mapperService, iw -> {
@@ -130,7 +132,7 @@ public class DiversifyingParentBlockQueryTests extends MapperServiceTestCase {
                 for (var doc : topDocs.scoreDocs) {
                     var entry = expectedTopDocs.get(i).pollFirstEntry();
                     assertNotNull(entry);
-                    assertThat(doc.score, equalTo(entry.getKey()));
+                    assertThat((double) doc.score, closeTo(entry.getKey(), 1e-5));
                     var storedDoc = storedFields.document(doc.doc, Set.of("id"));
                     assertThat(storedDoc.getField("id").binaryValue().utf8ToString(), equalTo(entry.getValue()));
                 }
@@ -143,9 +145,9 @@ public class DiversifyingParentBlockQueryTests extends MapperServiceTestCase {
             builder.startObject();
             builder.field("id", id);
             builder.startArray("nested");
-            for (int i = 0; i < vectors.length; i++) {
+            for (float[] vector : vectors) {
                 builder.startObject();
-                builder.field("emb", vectors[i]);
+                builder.field("emb", vector);
                 builder.endObject();
             }
             builder.endArray();

